master/src/main/scala/org/apache/celeborn/service/deploy/master/tags/TagsManager.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.service.deploy.master.tags
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters.{asScalaIteratorConverter, mapAsScalaConcurrentMapConverter}
+
+import org.apache.celeborn.common.internal.Logging
+import org.apache.celeborn.common.meta.WorkerInfo
+import org.apache.celeborn.service.deploy.master.tags.TagsManager.{Tag, TagsStore}
+
+object TagsManager {
+  type Tag = String
+  type WorkerId = String
+
+  type TagsStore = ConcurrentHashMap[Tag, Set[WorkerId]]
+}
+
+class TagsManager extends Logging {
+  private val tagStore = new TagsStore()
+
+  def getTaggedWorkers(tag: Tag, workers: List[WorkerInfo]): List[WorkerInfo] = {
+    if (!tagStore.containsKey(tag)) {
+      logWarning(s"Tag $tag not found in cluster")
+      return List.empty
+    }
+
+    workers.filter(worker => tagStore.get(tag).contains(worker.host))
+  }
+
+  def addTagToWorker(tag: Tag, worker: WorkerInfo): Unit = {
+    val workerId = worker.host
+    val workers = tagStore.get(tag)
+
+    logInfo(s"Adding Tag $tag to worker $workerId")
+    if (workers == null) {
+      tagStore.put(tag, Set(workerId))
+    } else {
+      tagStore.put(tag, workers + workerId)
+    }
+  }
+
+  def removeTagFromWorker(tag: Tag, worker: WorkerInfo): Unit = {
+    val workerId = worker.host
```

Comment:
The worker's host cannot be used as the unique identifier of a worker. It is more appropriate to use the WorkerInfo#toUniqueId method.

Reply:
Or we can just put the WorkerInfo, it's comparable.

Reply:
@zwangsheng Thats a good point but `WorkerInfo#toUniqueId` contains too much information. IMO for the external interface like File / DB / API passing the worker unique id can become cumbersome to manage for admins while host names are easy to manage.

Another option to uniquely identify a worker can be `workerHost:rpcPort` which will also be unique.

Reply:
@s0nskar Sorry for the late reply, using just host may cause problems in some users' case.

Use `workerHost:rpcPort` as the unique key seems reasonable to me

Reply:
updated the PR to use `toUniqueId` as it is predefined. 

Filed a follow up task to decide if we should use `workerHost:rpcPort` or something better.